src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.component.ts

```diff
@@ -549,21 +542,22 @@ export class OnlineGameWrapperComponent extends GameWrapper implements OnInit, O
     public setPlayersDatas(updatedICurrentPart: Part): void {
         display(OnlineGameWrapperComponent.VERBOSE, { OnlineGameWrapper_setPlayersDatas: updatedICurrentPart });
         this.players = [
-            updatedICurrentPart.doc.playerZero,
-            updatedICurrentPart.doc.playerOne];
+            MGPOptional.of(updatedICurrentPart.doc.playerZero),
+            MGPOptional.ofNullable(updatedICurrentPart.doc.playerOne),
+        ];
         assert(updatedICurrentPart.doc.playerOne != null, 'should not setPlayersDatas when players data is not received');
-        this.currentPlayer = this.players[updatedICurrentPart.doc.turn % 2];
-        let opponentName: string = '';
-        if (this.players[0] === this.getPlayerName()) {
+        this.currentPlayer = this.players[updatedICurrentPart.doc.turn % 2].get();
+        let opponentName: MGPOptional<string> = MGPOptional.empty();
+        if (this.players[0].equalsValue(this.getPlayerName())) {
             this.observerRole = Player.ZERO.value;
-            opponentName = this.players[1];
-        } else if (this.players[1] === this.getPlayerName()) {
+            opponentName = Utils.getNonNullable(this.players[1]);
+        } else if (this.players[1].equalsValue(this.getPlayerName())) {
             this.observerRole = Player.ONE.value;
-            opponentName = this.players[0];
+            opponentName = Utils.getNonNullable(this.players[0]);
```

Comment:
Ni this.players ni this.players[0] ne sont nullable, le second est juste optionnel !

Reply:
good catch, fixed